api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -436,4 +363,65 @@ private DbConceptSet getDbConceptSet(
     }
     return conceptSet;
   }
+
+  public static ConceptSet toClientConceptSet(DbConceptSet dbConceptSet) {
+    return new ConceptSet()
+        .etag(Etags.fromVersion(dbConceptSet.getVersion()))
+        .lastModifiedTime(dbConceptSet.getLastModifiedTime().getTime())
+        .creationTime(dbConceptSet.getCreationTime().getTime())
+        .description(dbConceptSet.getDescription())
+        .id(dbConceptSet.getConceptSetId())
+        .name(dbConceptSet.getName())
+        .domain(dbConceptSet.getDomainEnum())
+        .participantCount(dbConceptSet.getParticipantCount())
+        .survey(dbConceptSet.getSurveysEnum())
+        .creator(
+            dbConceptSet.getCreator() != null ? dbConceptSet.getCreator().getUsername() : null);
+  }
+
+  private ConceptSet toClientConceptSetWithConcepts(DbConceptSet dbConceptSet) {
+    ConceptSet conceptSet = toClientConceptSet(dbConceptSet);
+    if (!dbConceptSet.getConceptIds().isEmpty()) {
+      Iterable<DbConcept> concepts = conceptService.findAll(dbConceptSet.getConceptIds());
```

Comment:
Ideally the `DbConceptSet` should handle finding an empty collection and return an empty `Iterable`. It makes me a little bit nervous to see the call site have to do basic checks like this.